include/libtorrent/packet_pool.hpp

```diff
@@ -115,17 +124,18 @@ namespace libtorrent
 
 		packet_ptr alloc()
 		{
-			if (m_storage.empty())
-				return packet_ptr(create_packet(allocate_size));
-			else
-			{
-				auto ret = std::move(m_storage.back());
-				m_storage.pop_back();
-				return ret;
-			}
+			if (m_storage.empty()) return packet_ptr(create_packet(allocate_size));
+			auto ret = std::move(m_storage.back());
+			m_storage.pop_back();
+			return ret;
+		}
+
+		void decay()
+		{
+			if (m_storage.empty()) return;
```

Comment:
I remove one packet from here per second, meaning that it will be emptied 10 seconds after a burst ends